### PR TITLE
Handle cases when fullCut or partCut not available

### PR DESCRIPTION
**Pull request by katyukha**
### Contributor checklist

- [x] I have read the CONTRIBUTING.rst
- [x] I have tested my contribution on these devices:
 * e.g. NETUM NT-5890K
- [ ] My contribution is ready to be merged as is

----------

### Description

Related [PR](https://github.com/receipt-print-hq/escpos-printer-db/pull/25) introduces new features:
- paperPartCut
- paperFullCut

This PR handles them with minimal code changes.

### CODE REVIEW

File: src/escpos/escpos.py [hunk 1]
```diff
@@ -578,9 +578,11 @@ def cut(self, mode=''):
         # Fix the size between last line and cut
         # TODO: handle this with a line feed
         self._raw(b"\n\n\n\n\n\n")
-        if mode.upper() == "PART":
+
+        if mode.upper() == "PART" and self.profile.supports('paperPartCut'):
             self._raw(PAPER_PART_CUT)
-        else:  # DEFAULT MODE: FULL CUT
+        elif mode.upper() != "PART" and self.profile.supports('paperFullCut'):
```
Comment (patkan) [inline src/escpos/escpos.py]: You probably mean "FULL".

Comment (patkan) [inline src/escpos/escpos.py]: Moreover the full cut should also happen in the case that `mode` is not supplied.

For the case that `paperFullCut` is not available but `paperPartCut` is, defaulting to a partial cut would be good.

**Comment by katyukha** [comment]
Refactored `cut` method. Also it now uses new `print_and_feed` method, which implements `ESC d n` command ([doc](http://content.epson.de/fileadmin/content/files/RSD/downloads/escpos.pdf) and [doc2](http://www.starmicronics.com/support/mannualfolder/escpos_cm_en.pdf) used for it).
Is it right way, or it is better to use old code (using `\n\n\n\n\n\n` str instead)?

File: src/escpos/escpos.py (hunk 1 above)
Comment (katyukha) [inline src/escpos/escpos.py]: Seems to be implemented, to use one available method. But i am not sure if it is right to use FullCut if requested PartCut. May be just ignore it, if PartCut not available on printer?

**Comment by katyukha** [comment]
Related PR [receipt-print-hq/escpos-printer-db/pull/25](https://github.com/receipt-print-hq/escpos-printer-db/pull/25) is merged. code updated.

Is it Ok?

**Comment by patkan** [comment]
Hi @katyukha thank you!
Looks good to me. I am not sure about some names, but I think it is best to nail those down when we have to touch the complete API for the v3-release.

Just one last idea: How about issuing a `LF` instead of `ESC d n` in the case of 1 line? I don't know the level of support of this command and this way we would ensure that printing at least one line works.

**Comment by katyukha** [comment]
This commit fixes bug in `control` method, which makes using `control('LF')` have not result.

Also, it seems that `cut` command here is implemented incorrectly due to [ESC/POS doc](http://content.epson.de/fileadmin/content/files/RSD/downloads/escpos.pdf)

Here is following command code is used:

```python
# Paper Cutter
_CUT_PAPER = lambda m: GS + b'V' + m
PAPER_FULL_CUT = _CUT_PAPER(b'\x00')  # Full cut paper
PAPER_PART_CUT = _CUT_PAPER(b'\x01')  # Partial cut paper
```

But, as documentation says, there are two possible functions signatures for `cut`:

Function A: ```GS V m```
Function B: ```GS V m n```

Where ```m``` is:
- (Function A) m = 0, 1, 48, 49
- (Function B m = 65, 66

And ```n``` for function B is number of lines to feed.

So for **Function A**, possible commands:

```python
_CUT_PAPER = lambda m: GS + b'V' + m
PAPER_FULL_CUT = _CUT_PAPER(b'\x00\x30')  # Full cut paper
PAPER_PART_CUT = _CUT_PAPER(b'\x01\0x31')  # Partial cut paper
```

And for **Function B**, possible commands:

```python
_CUT_PAPER = lambda m, n=1: GS + b'V' + m + six.int2byte(n)
PAPER_FULL_CUT = _CUT_PAPER(b'\x41', n)  # Full cut paper
PAPER_PART_CUT = _CUT_PAPER(b'\x42', n)  # Partial cut paper
```

Also, may be it is a good idea to add number of lines to feed as parametr to cut?

As for names? You mean rename method `print_and_feed` for example to `feed`?

**Comment by patkan** [comment]
Hi @katyukha, sorry for taking so long. I wanted to merge your contribution, but I've got problems with your fix to `cut`. For my TM-T88II this does not work. It will print a 0 or 1 when cutting. Are you sure with the `\x31`. Where does this come from? You cite the reference but I can't find a mention of said sequence. (for cut it is e.g. just `GS V 0` according to the reference for mode A)

If this is needed for your printer, then we will have to add it as a capability. In this case I would like to do this in a different PR to get this one merged.

**Comment by katyukha** [comment]
Hi @patkan ! Yes, last commit seems to be incorrect( So i will revert it back.

**Comment by patkan** [comment]
Merged, thank you very much!

**Pull request closed without merging**